vnext/Microsoft.ReactNative/JsiApi.h

```diff
@@ -50,6 +50,22 @@ struct JsiError : JsiErrorT<JsiError> {
   std::optional<facebook::jsi::JSINativeException> const m_nativeException;
 };
 
+// Wraps up the IJsiHostObject
+struct HostObjectWrapper : facebook::jsi::HostObject {
+  HostObjectWrapper(Microsoft::ReactNative::IJsiHostObject const &hostObject) noexcept;
+
+  facebook::jsi::Value get(facebook::jsi::Runtime &runtime, const facebook::jsi::PropNameID &name) override;
```

Comment:
>) [](start = 97, length = 1)

(nit) do we care about adding noexcept here?

Reply:
When we override methods from the base facebook::jsi::HostObject interface we must not add the noexcept specification. It is mostly because in JSI we communicate errors using exceptions. If host object has an error then it is expected that the method implementation throws an exception. Then we intercept the exception and convert it to the ABI-safe error passing.